src/components/Single-product-detail/SingleProductDetails.jsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import "./singleProductDetails.css";
+import { useParams } from "react-router-dom";
+import { Carousel } from "../Carousel/Carousel";
+import { useCart } from "../../contexts/cart-context";
+import { useWishlist } from "../../contexts/wishlist-context";
+import { getOriginalPrice } from "../../utils/productutilFuncs";
+import { useProductsDetail } from "../../contexts/productsDetail-context";
+
+const SingleProductDetails = () => {
+
+    const { productId } = useParams();
+    const { productsDataFetchedFromApi: productsData } = useProductsDetail();
+    const { stateOfWishlist, dipatchOfWishlist } = useWishlist();
+    const { stateOfCart, dispatchOfCart } = useCart();
+
+    const getTheProductDetails = (productsData, productId) => {
+         return productsData.find(product => product._id === productId)
+    }
+
+    const singleProduct = getTheProductDetails(productsData, productId)
+
+    return (
+        <div className="ap_single-product-detail-component">
+            <div className="ap_single-product-container">
+                <div className="ap_singleProduct-carousel-container">
+                    <Carousel>
+                        {
+                            singleProduct.img.map((productImg, index) => (
+                                <div className="ap_single-product-img" key={index}>
+                                    <img src={ productImg } />
```

Comment:
you can add `alt` attribute for image tag

Reply:
yes, will take care of safe fails as well!! Thanks for pointing it out!!